Remotion/ObjectBinding/Web.ClientScript/Scripts/BocAutoCompleteReferenceValue.UI.ts

```diff
@@ -1809,11 +1809,13 @@ namespace Remotion.BocAutoCompleteReferenceValue
             maxHeight = '';
         }
 
-        const availableWidth = position.left + LayoutUtility.GetOuterWidth(reference);
-        const minWidth = LayoutUtility.GetOuterWidth(reference);
-        const maxWidth = Math.min (isNaN (options.maxWidth) ? LayoutUtility.GetOuterWidth(reference) : options.maxWidth, availableWidth);
-        const requiredWidth = contentWidth + 30;
-        const elementWidth = Math.max (Math.min (requiredWidth, maxWidth), minWidth);
+        const referenceWidth = LayoutUtility.GetOuterWidth(reference);
+        const bufferFromLeft = 30;
+        const availableWidth = position.left + referenceWidth - bufferFromLeft;
+        const maxWidth = Math.min (isNaN (options.maxWidth) ? referenceWidth : options.maxWidth, availableWidth);
+        const requiredWidth = contentWidth + 1; //to make sure that the inaccurate content width calculation is accurate enough
+        const maxAllowedWidth = Math.min(requiredWidth, maxWidth)
+        const elementWidth = Math.max(referenceWidth, maxAllowedWidth)
```

Comment:
Okay, while testing this, I may have found the reason for the original `+ 30`. we may not actually have cared about having a left-margin. Instead, the 30px where meant to ensure whitespace for the scrollbar. Now, if you type `a`, you will see that there's a horizontal scrollbar instead of the popup opening up further to the left.

What would be needed to incorporate correct scollbar capacity? Basically, if the popup using overflow:hidden, no space is reserved and if overflow:auto, we need to reserve some space for the scrollbar. 20px is probably enough, should be named scrollbarWidth or some such (no more magic numbers).

```
.ac_informationPopUp div
{
  /* overflow is not specificed, we can change this to explicitly 'visible' if need be. */
  /* overflow: visible; */
}
.ac_results div
{
  overflow: auto;
}

```

With this new discovery, we can decide to ignore the +30 in line 1813. I don't think the margin was the original intent, just a misinterpretation of the code. Of course, it could still look okay with the new semantics.

Reply:
We might be able to use https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-gutter during measuring to include the scrollbar gutter in the size calculation